go-runtime/compile/schema.go

```diff
@@ -603,6 +660,133 @@ func parseDatabaseDecl(pctx *parseContext, node *ast.CallExpr, dbType string) {
 	pctx.module.Decls = append(pctx.module.Decls, decl)
 }
 
+// parseTopicDecl expects: _ = ftl.RegisterTopic[EventType]("name_literal")
+func parseTopicDecl(pctx *parseContext, node *ast.CallExpr) {
+	// already extracted topic in the initial pass of the ast graph
+	// we did not do event type resolution yet, so we need to do that now
+	name, nameErr := topicNameFromCallExpr(node)
+	if nameErr != nil {
+		// error already added in previous pass
+		return
+	}
+
+	var topic *schema.Topic
+	for _, d := range pctx.module.Decls {
+		if d, ok := d.(*schema.Topic); ok && d.Name == name {
+			topic = d
+		}
+	}
+
+	// update topic's event type
+	indexExpr, ok := node.Fun.(*ast.IndexExpr)
+	if !ok {
+		pctx.errors.add(errorf(node, "must have an event type as a type parameter"))
+		return
+	}
+	typeParamType, ok := visitType(pctx, node.Pos(), pctx.pkg.TypesInfo.TypeOf(indexExpr.Index), topic.Export).Get()
+	if !ok {
+		pctx.errors.add(errorf(node, "invalid event type"))
+		return
+	}
+	topic.Event = typeParamType
+}
+
+// parseSubscriptionDecl expects: var _ = ftl.RegisterSubscription(topicHandle, "name_literal")
+func parseSubscriptionDecl(pctx *parseContext, node *ast.CallExpr) {
+	var name string
+	var topicRef *schema.Ref
+	if len(node.Args) == 2 {
+		if topicIdent, ok := node.Args[0].(*ast.Ident); ok {
```

Comment:
There should be a function that already does this...resolves an expression to a reference.

Reply:
`parseSelectorRef()` does the trick 👍

Reply:
Hmm no it doesn't